test/es5/freeze.js

```diff
@@ -3,47 +3,117 @@
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
 
-function write(args)
-{
- WScript.Echo(args);
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+  this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
 }
 
-write("TestCase1");
-write(Object.freeze.length);
-write(Object.isFrozen({}));
-
-write("TestCase2 - freeze & add a property");
-var a = {x:20, y:30};
-Object.freeze(a);
-SafeCall(function() { a.z = 50; });
-write(Object.getOwnPropertyNames(a));
-write(Object.isFrozen(a));
-
-write("TestCase3 - freeze & delete a property");
-var a = {x:20, y:30};
-Object.freeze(a);
-SafeCall(function() { delete a.x; });
-write(Object.getOwnPropertyNames(a));
-write(Object.isFrozen(a));
-write(a.x);
-
-write("TestCase4 - freeze & modify a property");
-var a = {x:20, y:30};
-Object.freeze(a);
-SafeCall(function() { a.x = 40; });
-SafeCall(function() { a.y = 60; });
-write(Object.getOwnPropertyNames(a));
-write(Object.isFrozen(a));
-write(a.x);
-
-function SafeCall(f)
-{
-  try
+var tests = [
   {
-    f();
-  }
-  catch (e)
+      name: "Add, delete, modify properties after freezing",
+      body: function () {
+          let a = {x: 42};
+          
+          Object.freeze(a);
+          assert.isFalse(Object.isExtensible(a));
+          assert.isTrue(Object.isSealed(a));
+          assert.isTrue(Object.isFrozen(a));
+
+          // cannot add new properties
+          a.y = 17;
+          assert.isFalse(a.hasOwnProperty('y'));
+          assert.throws(function () { 'use strict'; a.y = 17; }, TypeError, 
+            "Should throw on creating a new property in frozen object in strict mode", 
+            "Cannot create property for a non-extensible object");
+
+          // cannot delete properties
+          delete a.x;

```

Comment:
may want to assert that this returns false.

Reply:
done